### ISSUE TITLE
USB host controller not probed with v5.12-spx-dt and v5.13-spx-dt

### ISSUE THREAD

**Issue by jhaxhiaj** [linux-surface/kernel#121]
I'm trying to run Linux on Surface Pro X SQ2.

With ACPI, the kernel probes correctly the USB Host Controller dwc3-qcom and I can connect a USB keyboard and and USB Network dongle. 

While booting with v5.12-spx-dt the device tree , the kernel doesn't probe usb host controllers but manages to mount the roofs from initrd.
v5.13-spx-dt  with the dt doesn't boot for some reasons, the console starts printing then the screen gets black.

**Comment by qzed** [comment]
Yeah, the `-dt` branches have some issues... IIRC USB on the USB-C ports should work? USB for the dock definitely doesn't work. Not sure anymore though. Still very much a work in progress to which I hope to come back to in a month or so.

**Comment by jhaxhiaj** [comment]
> IIRC USB on the USB-C ports should work

This was my understanding from the discussions there https://github.com/Sonicadvance1/linux/issues/27 .
For some reason, the USB host controllers aren't even probed. 
I'm starting to suspect the device tree. 

I ran the following commands from grub:
```
devicetree /root/sc8180x-surface-pro-x.dtb
linux /root/Image.gz efi=novamap
initrd /root/rootfs.cpio.gz 
boot 
```

**Comment by jhaxhiaj** [comment]
![Uploading 8C80BDE5-B538-4A3B-AC67-BC0FB3DE5E1E.jpeg…]()

**Comment by jhaxhiaj** [comment]
![8A115934-D2F7-4FAE-A9AA-9A19C94AFEDF](https://user-images.githubusercontent.com/101270199/164261868-fba68e36-4af1-4e13-86e0-426a19b75ced.jpeg)

I added the capture of boot logs

**Comment by qzed** [comment]
Thanks! I'll try to have a look at that when I have a bit more time.

**Comment by qzed** [comment]
Okay, so I still haven't gotten around to recompile a new kernel for the SPX but here's a config from a working v5.13-spx-dt one (not sure if it's the exact same state of the branch though, it's been laying around a while): [config.txt](https://github.com/linux-surface/kernel/files/8598441/config.txt). USB devices at least show up in lsusb and as far as I can tell everything seems to be functional.

Before I do all that, I want to figure out what makes that one Grub version work... I don't really like relaying on that one built executable without having the means to reproduce it somehow. So if you have any hints (pointers to source/patches) that would be very helpful.

**Comment by jhaxhiaj** [comment]
I managed to make it boot by setting ``CONFIG_SC_GCC_8180X=Y`` and adding ``clk_ignore_unused`` to cmdline. 
Without ``clk_ignore_unused ``,  Linux don't finish the boot process. I'm not sure whether it's a bug on the clock controller or the clock is not correctly described in the device tree and get disabled.

**Comment by qzed** [comment]
Hmm, I didn't have to set that option. Regarding GRUB, I've written down some stuff [here](https://github.com/linux-surface/grub/issues/1).

**Comment by qzed** [comment]
To come back to this: I've noticed that I did include `gcc-sc8180x` in the initram... which explains why that worked as module for me, also makes sense as that's pretty essential. I built that as module because as you've said there's some issue during boot that locks up the device.

As it does work built-in and with the `clk_ignore_unused` option, it looks like something is trying to access an unclocked register. Specifically at some time during boot. I suspect that when the driver is build as module, it loads later, at which point the clock may be requested by something else and thus everything works. I'm not sure yet how to debug this but it's definitely something that should be fixed in the DT or kernel.
